### ISSUE TITLE
Documentation restructuring (core submodule)

### ISSUE THREAD

**Issue by bmcfee** [librosa/librosa#1478]
The documentation site is generally pretty navigable, but I think the core submodule (sub-heading "Core IO and DSP") is both too large and at this point, inaccurately named.

At present, it looks as follows:
![image](https://user-images.githubusercontent.com/1190540/166687856-7a1c3198-59ef-472d-86d6-2ed223d8d365.png)
where each of those sub-sub-headings has somewhere between 2 and 12 function entries.

I think we can easily break this up into some more meaningful sections, which can be promoted to the top level of the index.  Originally, I had designed the documentation index to reflect the submodule structure of the package, but I don't think that's terribly relevant anymore.

I propose that we eliminate the "Core IO and DSP" top-level heading, and replace it with the following:

- Signals  (includes load, stream, resample, and any signal generators and helper utilities)
- Spectral representations
- Magnitude scaling
- Unit conversion (time, frequency)
- Music notation
- Pitch and tuning
- Harmonics
- Phase recovery
- Miscellaneous (collects current misc + frequency range generators)

I'm also open to other suggestions for top-level organization, but the above seems at least plausible and a bit easier to navigate IMO.

Are there other sections of the docs that could benefit from restructuring?

**Comment by dana-gill** [comment]
Hey Brian -- Dana here (my old @danagilliann is no longer accessible) 😄  I'm happy to start working on this. I can start of with

> I propose that we eliminate the "Core IO and DSP" top-level heading, and replace it with the following:

**Comment by bmcfee** [comment]
:wave: Thanks @dana-gill , that'd be great!

**Comment by dana-gill** [comment]
I do have to admit: Building the documentation is not very easy since the documentation for it was a bit scattered 😅  Would it make sense to propose the following issues?

1. Have a `Contributing` section within the sidebar. I think the easiest would be to just link it to the [Contributing Page on GitHub](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/CONTRIBUTING.md). Maybe it could be converted to `.rst` some time in the future.
2. Reference [Compiling Documentation](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/docs/BUILD.md) in the [Contributing Page](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/CONTRIBUTING.md) that way there's a clear path for those working on the documentation. I thought about moving the content of the Compiling Documentation into the Contributing Page, but since the `Makefile` does more than build the website, I think it's fine staying there

Let me know what you think :)

**Comment by bmcfee** [comment]
> Building the documentation is not very easy since the documentation for it was a bit scattered 😅

Yeah, I know, sorry for that!  It's a complicated process because we need to maintain historical doc builds, and readthedocs isn't viable for us due to the computational overhead of running all of the examples.



> Have a `Contributing` section within the sidebar. I think the easiest would be to just link it to the [Contributing Page on GitHub](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/CONTRIBUTING.md). Maybe it could be converted to `.rst` some time in the future.

If there's a way to auto-include the doc, rather than linking or converting it, that would be better.  Github wants it to be CONTRIBUTING.md (though other formats might be supported) so that it's discoverable and checks off the community guidelines box.



> 2\. Reference [Compiling Documentation](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/docs/BUILD.md) in the [Contributing Page](https://github.com/librosa/librosa/blob/8dae20c66353190749486abf6b332d93c4ab2b46/CONTRIBUTING.md) that way there's a clear path for those working on the documentation. I thought about moving the content of the Compiling Documentation into the Contributing Page, but since the `Makefile` does more than build the website, I think it's fine staying there

Yeah, that's a good idea.  It could also be expanded a bit to cover the doc linting and link checking we added to the CI so that people can run that locally if they want.

**Comment by dana-gill** [comment]
Hi @bmcfee! I created a branch called `documentation_sidebar_restructuring` with a preview of a possible solution to restructure the documentation side bar. Before I completely proceed with a full makeover of the sidebar, could you let me know if the general idea was what you had in mind? Thanks! 🌻

**Comment by bmcfee** [comment]
:wave: @dana-gill sorry this got buried in my notifications (I'm phasing back into full-time this week).  But yeah, this looks like what I had in mind - thanks!
